Sources/Commands/SwiftPackageRegistryTool.swift

```diff
@@ -0,0 +1,185 @@
+        var global: Bool = false
+
+        @Option(help: "Associate the registry with a given scope")
+        var scope: String?
+
+        // TODO: Uncomment once .netrc management is implemented
+
+        // @Option(help: "Specify a user name for the remote machine")
+        // var login: String?
+
+        // @Option(help: "Supply a password for the remote machine")
+        // var password: String?
+
+        @Argument(help: "The registry URL")
+        var url: String
+
+        func run(_ swiftTool: SwiftTool) throws {
+            guard let url = URL(string: self.url),
+                  url.scheme == "https"
+            else {
+                throw RegistryConfigurationError.invalidURL(self.url)
+            }
+
+            // TODO: Require login if password is specified
+
+            let path = try swiftTool.getRegistryConfigurationPath(fileSystem: localFileSystem, global: global)
+
+            var configuration = RegistryConfiguration()
+            if localFileSystem.exists(path) {
+                configuration = try RegistryConfiguration.readFromJSONFile(in: localFileSystem, at: path)
+            }
+
+            if let scope = scope {
+                configuration.scopedRegistries[scope] = .init(url: url)
+            } else {
+                configuration.defaultRegistry = .init(url: url)
+            }
+
+            try configuration.writeToJSONFile(in: localFileSystem, at: path)
+
+            // TODO: Add login and password to .netrc
+        }
+    }
+
+    struct Unset: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Remove a configured registry")
+
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+
+        @Flag(help: "Apply settings to all projects for this user")
+        var global: Bool = false
+
+        @Option(help: "Associate the registry with a given scope")
+        var scope: String?
+
+        func run(_ swiftTool: SwiftTool) throws {
+            let path = try swiftTool.getRegistryConfigurationPath(fileSystem: localFileSystem, global: global)
+            var configuration = try RegistryConfiguration.readFromJSONFile(in: localFileSystem, at: path)
+
+            if let scope = scope {
+                guard let _ = configuration.scopedRegistries[scope] else {
+                    throw RegistryConfigurationError.missingScope(scope)
+                }
+                configuration.scopedRegistries.removeValue(forKey: scope)
+            } else {
+                guard let _ = configuration.defaultRegistry else {
+                    throw RegistryConfigurationError.missingScope()
+                }
+                configuration.defaultRegistry = nil
+            }
+
+            try configuration.writeToJSONFile(in: localFileSystem, at: path)
+        }
+    }
+}
+
+// MARK: -
+
+private extension Decodable {
+    static func readFromJSONFile(in fileSystem: FileSystem, at path: AbsolutePath) throws -> Self {
```

Comment:
we typically prefer to put the visibility modifier on the method and not the extension

Reply:
I personally like ACL on `private` and `fileprivate` extensions, as it prevents helper APIs from leaking out to `internal`. But I'm happy to make this change if you feel strongly about it.

The only hitch is that the current `.swiftformat` configuration falls back to the default [`extensionAccessControl` rule](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#extensionaccesscontrol) behavior of putting ACL on extensions rather than declarations. If we're still using this tool, should we update that to match project conventions?